api/client/commands.go

```diff
@@ -2587,3 +2589,107 @@ func (cli *DockerCli) CmdExec(args ...string) error {
 
 	return nil
 }
+
+type containerStats struct {
+	Name             string
+	CpuPercentage    float64
+	Memory           float64
+	MemoryLimit      float64
+	MemoryPercentage float64
+	NetworkRx        float64
+	NetworkTx        float64
+}
+
+func (cli *DockerCli) CmdStats(args ...string) error {
+	cmd := cli.Subcmd("stats", "CONTAINER", "Stream the stats of a container", true)
+	cmd.Require(flag.Min, 1)
+	utils.ParseFlags(cmd, args, true)
+
+	m := &sync.Mutex{}
+	cStats := map[string]containerStats{}
+	for _, name := range cmd.Args() {
+		go cli.streamStats(name, cStats, m)
+	}
+	w := tabwriter.NewWriter(cli.out, 20, 1, 3, ' ', 0)
+	for _ = range time.Tick(500 * time.Millisecond) {
+		fmt.Fprint(cli.out, "\033[2J")
+		fmt.Fprint(cli.out, "\033[H")
+		fmt.Fprintln(w, "CONTAINER\tCPU %\tMEM USAGE/LIMIT\tMEM %\tNET I/O")
+		m.Lock()
+		ss := sortStatsByName(cStats)
+		m.Unlock()
+		for _, s := range ss {
+			fmt.Fprintf(w, "%s\t%.2f%%\t%s/%s\t%.2f%%\t%s/%s\n",
+				s.Name,
+				s.CpuPercentage,
+				units.BytesSize(s.Memory), units.BytesSize(s.MemoryLimit),
+				s.MemoryPercentage,
+				units.BytesSize(s.NetworkRx), units.BytesSize(s.NetworkTx))
+		}
+		w.Flush()
+	}
+	return nil
+}
+
+func (cli *DockerCli) streamStats(name string, data map[string]containerStats, m *sync.Mutex) error {
+	m.Lock()
+	data[name] = containerStats{
+		Name: name,
+	}
+	m.Unlock()
+
+	stream, _, err := cli.call("GET", "/containers/"+name+"/stats", nil, false)
+	if err != nil {
+		return err
```

Comment:
We should probably at least do `delete(data, name)`. No point of leaving empty row in the output in case of an error. Ideally the whole command would error when there are no valid containers.